pkg/tplink_adapter.py

```diff
@@ -70,16 +61,33 @@ def start_pairing(self, timeout):
             if not self.pairing:
                 break
 
-            _id = 'tplink-' + dev.sys_info['deviceId']
-            if _id not in self.devices:
-                if isinstance(dev, SmartPlug):
-                    device = TPLinkPlug(self, _id, dev)
-                elif isinstance(dev, SmartBulb):
-                    device = TPLinkBulb(self, _id, dev)
-                else:
-                    continue
+            self._add_device(dev)
+
+    def _add_device(self, dev):
+        """
+        Add the given device, if necessary.
 
-                self.handle_device_added(device)
+        dev -- the device object from pyHS100
+        """
+        _id = 'tplink-' + dev.sys_info['deviceId']
+        if _id not in self.devices:
+            if isinstance(dev, SmartStrip):
```

Comment:
I think this would be clearer in its own block since the _id check above is expected to do nothing if the device is a smartstrip

Reply:
Good idea. Fixed.